runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/solver/pathfinding/IDAStarMM.java

```diff
@@ -0,0 +1,787 @@
+
+					swap(locBlank.getX(), locBlank.getY(), loc1.getX(), loc1.getY());
+					swap(loc1.getX(), loc1.getY(), loc2.getX(), loc2.getY());
+					swap(loc2.getX(), loc2.getY(), locVal.getX(), locVal.getY());
+				}
+			}
+			else if (locVal.getY() == locBlank.getY())
+			{
+				int diff = locBlank.getX() - locVal.getX();
+				if (diff == 1)
+				{
+					//Right
+
+					//Last piece
+					if (y == DIMENSION - 1)
+					{
+						Point loc1 = new Point(locVal.getX() - 1, locVal.getY() - 1);
+						Point loc2 = new Point(locVal.getX() - 1, locVal.getY());
+						Point loc3 = new Point(locVal.getX(), locVal.getY() - 1);
+						Point loc4 = new Point(locVal.getX() + 1, locVal.getY() - 1);
+
+						swap(locBlank.getX(), locBlank.getY(), locVal.getX(), locVal.getY());
+
+						swap(locVal.getX(), locVal.getY(), loc3.getX(), loc3.getY());
+						swap(loc3.getX(), loc3.getY(), loc1.getX(), loc1.getY());
+						swap(loc1.getX(), loc1.getY(), loc2.getX(), loc2.getY());
+						swap(loc2.getX(), loc2.getY(), locVal.getX(), locVal.getY());
+						swap(locVal.getX(), locVal.getY(), loc3.getX(), loc3.getY());
+						swap(loc3.getX(), loc3.getY(), loc1.getX(), loc1.getY());
+						swap(loc1.getX(), loc1.getY(), loc2.getX(), loc2.getY());
+						swap(loc2.getX(), loc2.getY(), locVal.getX(), locVal.getY());
+						swap(locVal.getX(), locVal.getY(), locBlank.getX(), locBlank.getY());
+						swap(locBlank.getX(), locBlank.getY(), loc4.getX(), loc4.getY());
+						swap(loc4.getX(), loc4.getY(), loc3.getX(), loc3.getY());
+						swap(loc3.getX(), loc3.getY(), loc1.getX(), loc1.getY());
+						swap(loc1.getX(), loc1.getY(), loc2.getX(), loc2.getY());
+						swap(loc2.getX(), loc2.getY(), locVal.getX(), locVal.getY());
+
+						return;
+					}
+
+					Point loc1 = new Point(locVal.getX() + 1, locVal.getY() + 1);
+					Point loc2 = new Point(locVal.getX(), locVal.getY() + 1);
+					Point loc3 = new Point(locVal.getX() - 1, locVal.getY() + 1);
+					Point loc4 = new Point(locVal.getX() - 1, locVal.getY());
+
+					swap(locBlank.getX(), locBlank.getY(), loc1.getX(), loc1.getY());
+					swap(loc1.getX(), loc1.getY(), loc2.getX(), loc2.getY());
+					swap(loc2.getX(), loc2.getY(), loc3.getX(), loc3.getY());
+					swap(loc3.getX(), loc3.getY(), loc4.getX(), loc4.getY());
+					swap(loc4.getX(), loc4.getY(), locVal.getX(), locVal.getY());
+				}
+				else if (diff == -1)
+				{
+					//Left
+					swap(locBlank.getX(), locBlank.getY(), locVal.getX(), locVal.getY());
+				}
+			}
+		}
+	}
+
+	private void swap(int x1, int y1, int x2, int y2)
+	{
+		PuzzleState newState = currentState.swap(x1, y1, x2, y2);
+
+		currentState = newState;
+		stateList.add(newState);
+	}
+
+	private Point findPiece(int val)
+	{
+		Point loc = null;
+
+		for (int x = 0; x < DIMENSION; x++)
+		{
+			for (int y = 0; y < DIMENSION; y++)
+			{
+				if (currentState.getPiece(x, y) == val)
+				{
+					return new Point(x, y);
+				}
+			}
+		}
+
+		return loc;
```

Comment:
This return is never reached, because if findPiece returns null your code crashes with an NPE.
It should never reach it, so maybe an exception should be thrown if the code for whatever reason gets there.